### PR TITLE
Fix generation of loss spec from output spec

### PR DESCRIPTION
**Pull request by kwen2501**
## Description

In PiPPy we automatically generate loss spec from output chunk spec.
Terminology:
"loss spec": a one-hot map indicating which output value is a loss. It consists of True, False values.
"output chunk spec": a data structure corresponding to output format describing which (chunked) value should be merged and which should be reduced (such as loss).

## Issue

In previous code, we only considered the case where the output chunk spec is a dictionary.
But in cases such as `return logits, loss`, the output chunk spec is a tuple, i.e. `(TensorChunkSpec(0), sum_reducer)`.

## Fix

Use `fx.node.map_aggregate` to generalize the auto generation.
It takes a lambda function that outputs True/False.

## Testing

torchrun --nproc-per-node 4 local_test_chunkspec.py

### CODE REVIEW

**Comment by yjhao** [comment]
Thanks for fixing this!

**Review by yjhao — APPROVED** [review]
